README.md

```diff
@@ -96,19 +96,22 @@ use hashbrown::HashMap;
 let mut map = HashMap::new();
 map.insert(1, "one");
 ```
-
+## Flags
 This crate has the following Cargo features:
 
-- `nightly`: Enables nightly-only features: `#[may_dangle]`.
+- `inline-more`: Adds inline hints to most functions, improving run-time performance at the cost
+   of compilation time. (enabled by default)
+- `ahash`: Compiles with ahash as default hasher. (enabled by default)
+- `ahash-run-time-rng`: Uses randomly generated keys for each hashmap to provide DOS resistance.
+   This requires the standard library. (disabled by default)
+- `ahash-compile-time-rng`: This is an alternative to `ahash-run-time-rng` that works by pre-generating keys at 
+   compile time and embedding them as constants. The avoids the dependency on the standard library but means the 
+   binary will be slightly different each time it is compiled. (disabled by default)
```

Comment:
A thought occurs: is it really necessary to have these two flags at all? In `hashbrown` we make no guarantees about DoS resistance so we can simply always use constant keys. I would expect anyone needing DoS resistance to use the `ahash` crate directly (enabling any features they need).

Reply:
That would work if we didn't need to think about version upgrades. A user could depend of HashBrown 0.9.1 and aHash 0.5.6 and set the flags on aHash that they want, and everything would work as expected. However, if they upgrade to HashBrown 0.9.2 and it happens to depend on aHash 0.6.0 and they don't update the version of their direct dependency on aHash then cargo will pull in two versions of aHash. HashBrown will use one without the flags set, and the other will be unused, unless the code is manually specifying the hasher each time. So suddenly the flags will have no effect in some places and they won't notice.

Reply:
Anyone relying on those flags should explicitly use `HashMap<K, V, AHasher>` where `AHasher` is imported from their version of the `ahash` crate, not the one hashbrown uses.

Reply:
What about the case where there is an intermediate library? For example: Suppose there is a library for data processing that depends on HashBrown. If the library is generic, and does not know if it will be used in a context that where `std` is available or if DOS resistance is required. 

From your above answer it sounds like such a library not depend on HashBrown. Is this the case? I was attempting to model this case on the way many crates handle `std` or `serde` where it's a flag which is propagated to the application via intermediate dependencies. 

Ideally there would be a way to do this without all the intermediate dependencies being "aware" or it or the end application being tied to a specific version of its dependencies. But I am not aware of any way to do that within Cargo.

Reply:
As thing are, there is no performance advantage to fixed keys. However when dos resistance is not a concern: AHasher has a method `short_finish()` which is a faster alternative to `finish` (at some cost to quality). It is mutually exclusive with specialization for primitives though.

Reply:
Maybe this could be exposed on the `ahash` side as two separate hasher types: one with dos resistance and one without. This would neatly solve the issue.

Reply:
I have created https://github.com/tkaitchuck/aHash/issues/58 to track that. (as doing that properly will take some time.)